### ISSUE TITLE
Bug on multilangs domain generate sitemap

### ISSUE THREAD

**Issue by testt23** [cebe/luya-module-sitemap#8]
I have found a bug on Sitemap module. when the site has 3 languages, 3 sitemap domains are not "updated" by domains, and only uses the default one.

Example: 
Default - en - site.com 
Douche - de - site.de
Italian - it - site.it

Sitemap generate only for site.com

This is problem for SEO and for Google.

**Comment by cebe** [comment]
You mean sitemap on all domains only contains pages for the default domain?

Have you set up your site using https://luya.io/guide/app-language#domain-mapping "Domain Mapping" headline? Or are you using another method?

**Comment by testt23** [comment]
That's right - for all domains, the system generates the default domain. I have set it right as it is in the manual

**Comment by cebe** [comment]
seems this can be solved by using `buildItemLink()` from Menu class:

https://github.com/luyadev/luya-module-cms/blob/9a74bc5869c365e50bc455c4860e3af8c6145d34/src/Menu.php#L583-L586

instead of just adding `Yii::$app->request->hostInfo` here:

https://github.com/cebe/luya-module-sitemap/blob/0d7da9ad0e224b0d5ce0b431a98e63b3e80a8121/src/controllers/SitemapController.php#L86-L87

@nadar can you confirm this is the correct fix?

**Comment by nadar** [comment]
buildItemLink sadly does not create an absolute url, therefore hostInfo must be prepend, this is also what the menu component does when accessing the absolute url: https://github.com/luyadev/luya-module-cms/blob/master/src/menu/Item.php#L471

**Comment by nadar** [comment]
But it will prepend to correct language based on the given configuration and circumstances.

**Comment by cebe** [comment]
@nadar how does the Menu component function you linked work with different host names configured for different language? as described here: https://luya.io/guide/app-language#domain-mapping

**Comment by nadar** [comment]
It will change the default language for the given host. So its actually not part of the menu component, it just changes default language. And for the language handling the composition component is responsible. Does that help?

**Comment by cebe** [comment]
Seems I need to dig into this in more detail, it is not relevant for any site I have right now, so I'd need to set up a test environment to see how to solve this issue.

**Comment by nadar** [comment]
I think we would just need a method which does return the host for a given language based on the host mapping information. I could create such a function in the LUYA core, please let me know if i can help.

**Comment by SOHELAHMED7** [comment]
For domains with language AND country  specific content;  de-de, de-at, de-ch; en-gb, en-us etc. the domain of the first matching `langShortCode`will be considered

For eg 
` Composition::$hostInfoMapping`

```php
'hostInfoMapping' => [
    'http://example.com' => ['langShortCode' => 'en', 'countryShortCode' => 'us'],
    'http://example.co.uk' => ['langShortCode' => 'en', 'countryShortCode' => 'uk'],
    'http://example.de' => ['langShortCode' => 'de', 'countryShortCode' => 'de'],
],
```

sitemap.xml will have  `http://example.com` first matching domain for `en` langugae i.e. `langShortCode` and not `http://example.co.uk`

PR: https://github.com/cebe/luya-module-sitemap/pull/14

EDIT - 
no link of http://example.co.uk and http://example.de will be included in sitemap, never.

**Comment by nadar** [comment]
@SOHELAHMED7 I just pushed the new function into the luya core dev-master: https://github.com/luyadev/luya/blob/master/core/web/Composition.php#L196-L220

**Comment by nadar** [comment]
> no link of http://example.co.uk and http://example.de will be included in sitemap, never.

@SOHELAHMED7 yes, because this information is missing in the cms_nav_item / cms_nav table - so at first step you could just resolve langShortCode.

**Comment by testt23** [comment]
to understand that if I make an upgrade to a sitemap, everything will be right or I'm wrong

**Comment by nadar** [comment]
@testt23 As there is no major first release, there could be some breakings changes - but it should not as i can see ;-)

**Comment by cebe** [comment]
I just merged #14, @testt23 please try it out and report back if this works for your setup now.
